ci/ci.py

```diff
@@ -465,7 +469,7 @@ def main(argv):
         release_channel = "editor-alpha"
         make_release = True
         engine_artifacts = args.engine_artifacts
-    elif branch and branch.startswith("DEFEDIT-"):
+    elif branch and (branch.startswith("DEFEDIT-") or get_target_branch() == "editor-dev"):
```

Comment:
For pull_request workflows, it seems like the target branch name determines the matching rule. So anything targeting `editor-dev` (or a `DEFEDIT-`-prefixed branch name) will use the editor-only workflow. It follows that we need to use `archived-stable` artifacts here, because editor-only does not build Bob.